### DISCUSSION TITLE
[Feature Request] Directional Access for Doors

### DISCUSSION THREAD

**Discussion by eurtod** [unitystation/unitystation#9572] [Ideas]
### There is no other issue with the same feature request.

- [X] I have searched the existing issues

### Why is this feature needed/good for the game?

I think it would make sense that some doors can be accessed when exiting but not when entering. This would beneficial for people trying to leave medical so that doctors don't have to stop what they're doing and let fixed patients out for example. Also, like letting people into maints from a secured department but wouldn't let them enter the maints from the department.

### Describe your feature in detail.

This would be a complex feature I understand, I also understand it likely isn't worth the time to create and the complexity it would add. But it would allow two separate access ways from some doors. Allowing for example everyone to enter a door from the left but only science enter from the right. I suppose this would also mean there would be two hacking ways of the door or something.

### User stories

As a patient, I want to be able to exit medical when I'm completed without needing a doctor to let me out.
As a security officer, I want the janitor to be able to leave security through the maintenance but not enter from inside maintenance.

### Describe alternatives you've considered.

Maybe some kind of temporary pass that different departments can give people for a limited time that they can then deactivate remotely at their discretion.
The downsides of this are how much it would change how doors and access are set up and the fact I'm not sure how hacking would interact with this.

**Comment by PerfectTangent** [comment]
feature requests for new ideas should use discussions. Converting this issue to one.
